### ISSUE TITLE
How to improve our model given diagnosis

### ISSUE THREAD

**Issue by marcwww** [marcotcr/lime#384]
Dear Marcotcr, really appreciate it for your excellent work.

I am doing some tests with the [text-classification example](https://marcotcr.github.io/lime/tutorials/Lime%20-%20basic%20usage%2C%20two%20class%20case.html). And I am interested in utilizing the diagnosis given by SP-LIME to improve the model.

The screenshot is as follows:
![image](https://user-images.githubusercontent.com/10141302/65253175-c6b05680-db2c-11e9-8f86-0100d8d2f6b5.png)

As you can see, this improvement procedure is referred to the LIME paper. I don't understand why the performance decreases after removing the overfitted tokens. 

Could you please guide me on this? Thanks a lot.

**Comment by marcotcr** [comment]
I don't quite understand what your line 16 means, but will answer assuming this is the accuracy on the test set for the model trained without Posting and Host.

If there is signal in Posting and Host (and there is), it is present in both the training and the test sets. Removing such signal gives you a model that does not rely on it (and thus will generalize better), but will of course perform worse when such signal is present.

It might be easier to think about the case of data leaking: imagine that you accidentaly added the token 'THISISATHEISM' to every example with `label=0`. If you leave this token on training an test, your accuracy will be 100% for whatever model you train. If you train a model after this token is removed, your accuracy on the test set will _decrease_, but you will have a better model.

**Comment by marcwww** [comment]
> I don't quite understand what your line 16 means, but will answer assuming this is the accuracy on the test set for the model trained without Posting and Host.
> 
> If there is signal in Posting and Host (and there is), it is present in both the training and the test sets. Removing such signal gives you a model that does not rely on it (and thus will generalize better), but will of course perform worse when such signal is present.
> 
> It might be easier to think about the case of data leaking: imagine that you accidentaly added the token 'THISISATHEISM' to every example with `label=0`. If you leave this token on training an test, your accuracy will be 100% for whatever model you train. If you train a model after this token is removed, your accuracy on the test set will _decrease_, but you will have a better model.

That exactly answers my question. Thanks a lot for your enlightening reply 😄.
